frontend/appflowy_flutter/lib/plugins/database/widgets/cell_editor/checklist_cell_editor.dart

```diff
@@ -339,29 +351,28 @@ class _NewTaskItemState extends State<NewTaskItem> {
         children: [
           const HSpace(8),
           Expanded(
-            child: TextField(
-              focusNode: widget.focusNode,
-              controller: _textEditingController,
-              style: Theme.of(context).textTheme.bodyMedium,
-              decoration: InputDecoration(
-                border: InputBorder.none,
-                isCollapsed: true,
-                contentPadding: const EdgeInsets.symmetric(
-                  vertical: 6.0,
-                  horizontal: 2.0,
+            child: CallbackShortcuts(
+              bindings: {
+                const SingleActivator(LogicalKeyboardKey.enter): () =>
+                    _createNewTask(context),
+              },
+              child: TextField(
+                focusNode: widget.focusNode,
+                controller: _textEditingController,
+                style: Theme.of(context).textTheme.bodyMedium,
+                maxLines: null,
+                decoration: InputDecoration(
+                  border: InputBorder.none,
+                  isCollapsed: true,
+                  contentPadding: const EdgeInsets.symmetric(
+                    vertical: 6.0,
+                    horizontal: 2.0,
+                  ),
+                  hintText: LocaleKeys.grid_checklist_addNew.tr(),
                 ),
-                hintText: LocaleKeys.grid_checklist_addNew.tr(),
+                onSubmitted: (_) => _createNewTask(context),
+                onChanged: (value) => setState(() {}),
```

Comment:
Can you add a comment here explaining why we need to `setState` when the text is changing?

Reply:
Of course

Reply:
Actually, I think creating a new field to keep the enable status would be better than using an empty callback.

For example: `setState(() => isCreateButtonEnabled = !textEditingController.text.isEmpty)`